### ISSUE TITLE
Error or ambiguity in explaining `watch` VS `computed`

### ISSUE THREAD

**Issue by fnick851** [vuejs/docs-next#512]
On this page: https://v3.vuejs.org/guide/computed.html#watchers, in the "Watchers" section, near the bottom it says: "None of that would be possible with a computed property."

However I think it is quite possible to do the exact same thing with a computed property with getter and setter. Here is a demo: https://github.com/fnick851/vue3-doc-watch-vs-computed/blob/master/src/App.vue

Is this an error or do I miss the point? At least I think there is some ambiguity.

**Comment by NataliaTepluhina** [comment]
@fnick851 good point! What do you think about adding changing 2 data properties in the watcher so `computed` won't be an alternative due to side effect?

**Comment by fnick851** [comment]
@NataliaTepluhina I'm not exactly clear what you have in mind, but I think there could be some cases that involve side effect where computed property cannot achieve the same behavior. Like, when you try to change the <title> of a HTML page, maybe there is  an example where only watcher can make it work. But in general, you can still have side effects in the getter/setter of a computer property, no?

I think the limitations of computed property are:
1. It only runs when it is referenced somewhere, while watcher always runs when the thing it watches changes.
2. Computed property does not have access to the previous/old value of its dependency, while watcher does.
So maybe there could be a case that involves these two points?

**Comment by NataliaTepluhina** [comment]
@fnick851 it's not recommended to include any side effects on `computed` (if you try it without getter/setter, there should be a warning)

I like an idea about old/new value, we could add a condition there 👍🏻

**Comment by fnick851** [comment]
@NataliaTepluhina  
After some experiments, I think my previous two statements regarding the differences between `watch` and `computed` are not true at all... 😅
1. Using Vue devtool I was able to see that a computed property always rerun even if it is not referenced anywhere in the app, as long as its dependency is changed and the cached value is invalid.
2. You can technically still access the "old" value when using the `computed` property setter.

So I think the main differences are 1. caching and 2. code style (side effects are not recommended in `computed`.) 
Maybe the doc should highlight that on this page?

**Comment by LinusBorg** [comment]
Your first point is incorrect. If nothing ever reads the computed value, it will never update. Since you read it when you check it in devtools, it updates at that moment.

This is what is called lazy evaluation of computed properties

**Comment by fnick851** [comment]
@LinusBorg  Thanks for the explanation!
